### PR TITLE
Add SREM / SSEM and DSRC library JSON Schemas

### LINKED ISSUE #183: Define SSEM contract

### LINKED ISSUE #184: Define SREM contract

### PR DESCRIPTION
**Pull request by mathieu1fb**
**What's new**;

JSON Schemas:
Added SREM and SSEM schemas, alongside a DSRC schema regrouping common definitions for both SREM and SSEM (to be completed for future needs).

closes #184 and closes #183 

---
**What to do**:

1. Read the new schemas and check them against [ETSI repository docs](https://forge.etsi.org/rep/ITS/asn1/is_ts103301/-/tree/release2/docs) for potential mistakes.
2. Evaluate and discuss the impact of the use of new DSRC "units" before merging.

### CODE REVIEW

**Comment by ymorin-orange** [comment]
> In another issue on mobility messages ( #137) it was asked to create directories for each message. Perhaps you can put each message in a directory (and that will test relative link to dsrc?)
>
> Nota: in the PR linked to the same issue, I have change the "type" field of the header in "message_type". If it is approved, it should be the case for this messages too (can be done also in a 2.0.O version?)

I think we should decide now what we do, either one of:
* we merge the SREM/SSEM schemas first, so thus they should conform to the current practices and the PR doing the renaming and version bump should be rebased and would have to also update the SREM/SSEM schemans, or
* we merge the renaming and version bump first, so thus the SREM/SSEM schemas should conform to the new practices.

The second option sounds more appealing to me, but this means the new practices must be properly documented so as to simplify those new schemas.

File: schema/srem_schema_1-0-0.json [hunk 1]
```diff
@@ -46,10 +46,31 @@
     "message": {
       "type": "object",
       "required": [
+        "protocol_version",
+        "station_id",
```
Comment (ymorin-orange) [inline schema/srem_schema_1-0-0.json:50]: What's the PDU header?

What's the difference between those fields, and the top-level `version` and `source_uuid` ones?

Why does this need to be a separate commit and can't be included in the first one?

File: schema/srem_schema_1-0-0.json [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        },
+        "requests": {
+          "type": "array",
+          "description": "List of SignalRequest entries.",
+          "minItems": 1,
+          "maxItems": 32,
+          "items": {
+            "type": "object",
+            "required": [
+              "signal_request"
```
[the remark this answers is not in the document]
Comment (Hugues360) [reply schema/srem_schema_1-0-0.json:94]: Ah, yes. right.
But the name of the field should not be just "request" (singular) instead of "signal_request"

[the remark this answers is not in the document]
Comment (mathieu1fb) [reply schema/srem_schema_1-0-0.json:94]: Indeed, it should be `request`! I'll fix that.

File: schema/srem_schema_1-0-0.json (hunk 1 above)
Comment (mathieu1fb) [reply schema/srem_schema_1-0-0.json:50]: This is common to all ETSI ITS messages, and we have always included it at the top level of the `message` object.
`version` and `source_uuid` are Orange-specific and found in our top-level header, while `protocol_version` and `station_id` are ETSI-specific and found in the `message` section (could be needed if someone needs to translate our JSON messages into ASN1 UPER).

As for the separate commit, it is indeed not needed, I'll fix it.

File: schema/srem_schema_1-0-0.json [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                "type": "object",
+                "description": "Used (as part of a request message) to request either a priority or a preemption service from a signalized intersection.",
+                "required": [
+                  "intersection_reference_id",
+                  "request_id",
+                  "request_type",
+                  "inbound_lane"
+                ],
+                "properties": {
+                  "intersection_reference_id": {
```
[the remark this answers is not in the document]
Comment (mathieu1fb) [reply schema/srem_schema_1-0-0.json:107]: You're right, I'll fix it.

File: schema/dsrc_schema_1-0-0.json [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+      "type": "integer",
+      "description": "16-bit globally unique identifier assigned to an entity responsible for assigning Intersection IDs in the region over which it has such authority. The value zero shall be used for testing, and should only be used in the absence of a suitable assignment. A single entity which assigns intersection IDs may be assigned several RoadRegulatorIDs. These assignments are presumed to be permanent.",
+      "minimum": 0,
+      "maximum": 65535
+    },
+    "intersection_reference_id": {
+      "type": "object",
+      "description": "Conveys the combination of an optional RoadRegulatorID and of an IntersectionID that is unique within that region.",
+      "required": [
+        "intersection_id"
```
[the remark this answers is not in the document]
Comment (mathieu1fb) [reply schema/dsrc_schema_1-0-0.json:72]: Only `id` indeed, I'll fix it.

File: schema/ssem_schema_1-0-0.json [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        },
+        "status": {
+          "type": "array",
+          "description": "List of SignalStatus entries.",
+          "minItems": 1,
+          "maxItems": 32,
+          "items": {
+            "type": "object",
+            "required": [
+              "signal_status"
```
[the remark this answers is not in the document]
Comment (ymorin-orange) [reply schema/ssem_schema_1-0-0.json:94]: Yes, agreed.

**Comment by ymorin-orange** [comment]
> I think we should decide now what we do, either one of:
>     * we merge the SREM/SSEM schemas first [...]
>     * we merge the renaming and version bump first [...]

We decided for the first option: merge SREM/SSEM schemas first.

**Review by ymorin-orange — APPROVED** [review]
